packages/graphqlsp/src/diagnostics.ts

```diff
@@ -147,39 +150,75 @@ export function getGraphQLDiagnostics(
           };
         }
 
-        const [typeQuery] = callExpression.typeArguments;
+        let foundNode,
+          foundFilename = filename,
+          ref,
+          start,
+          length;
+        if (callExpression.typeArguments) {
+          const [typeQuery] = callExpression.typeArguments;
+          start = typeQuery.getStart();
+          length = typeQuery.getEnd() - typeQuery.getStart();
 
-        if (!ts.isTypeQueryNode(typeQuery)) {
-          // Provide diagnostic about wroong generic
-          return {
-            category: ts.DiagnosticCategory.Warning,
-            code: MISSING_PERSISTED_TYPE_ARG,
-            file: source,
-            messageText:
-              'Provided generic should be a typeQueryNode in the shape of graphql.persisted<typeof document>.',
-            start: typeQuery.getStart(),
-            length: typeQuery.getEnd() - typeQuery.getStart(),
-          };
-        }
+          if (!ts.isTypeQueryNode(typeQuery)) {
+            return {
+              category: ts.DiagnosticCategory.Warning,
+              code: MISSING_PERSISTED_TYPE_ARG,
+              file: source,
+              messageText:
+                'Provided generic should be a typeQueryNode in the shape of graphql.persisted<typeof document>.',
+              start,
+              length,
+            };
+          }
+          const { node: found, filename: fileName } =
+            getDocumentReferenceFromTypeQuery(typeQuery, filename, info);
+          foundNode = found;
+          foundFilename = fileName;
+          ref = typeQuery.getText();
+        } else if (callExpression.arguments[1]) {
+          start = callExpression.arguments[1].getStart();
+          length =
+            callExpression.arguments[1].getEnd() -
+            callExpression.arguments[1].getStart();
+          if (
+            !ts.isIdentifier(callExpression.arguments[1]) &&
+            !ts.isCallExpression(callExpression.arguments[1])
+          ) {
+            return {
+              category: ts.DiagnosticCategory.Warning,
+              code: MISSING_PERSISTED_TYPE_ARG,
+              file: source,
+              messageText:
+                'Provided argument should be an identifier or invocation of "graphql" in the shape of graphql.persisted(hash, document).',
+              start,
+              length,
+            };
+          }
```

Comment:
This seems a bit optimistic, so we may have to leave a todo here?

Reply:
Realistically it can only be `graphql.persisted(hash, document || graphql())` no?

Reply:
Hard to tell, I'm mostly afraid of indirection, e.g. re-exports with reassignments, etc, messing with later checks. So, just realised this doesn't really have much to do with the check here, but with the extraction in the other place 🤔